docs/try/tutorials/geomodel/geomodel-basics.md

```diff
@@ -3,71 +3,94 @@
 geomodeler. This tutorial presents a simple code to load manipulate and export `GeoModel`.
 
+A part of the code associated to this tutorial is available in the `RINGMesh/doc/tutorials/src/geomodel-basic.cpp`.
+It can be compiled by selecting the `RINGMESH_WITH_TUTORIALS` option during the RINGMesh configuration with CMake. 
+
 ### GeoModel I/O
 
 GeoModel stores the geometry and the topology of a geological model. It can be imported from boundary
 representation or volumetric representation file.
 RINGMesh uses an extensible design to handle various file formats.
-The file extention is the key for selecting the suitable loader/saver.
+The file extension is the key for selecting the suitable loader/saver.
 The Input/Output file formats currently supported by RINGMesh are listed [here](/features/file_formats).
 
 A GeoModel can be loaded and saved as follow:
 
 ```c++
+//include for RINGMesh::GeoModel3D
+#include <ringmesh/geomodel/core/geomodel.h> 
+//include for RINGMesh::geomodel_load & RINGMesh::geomodel_save
+#include <ringmesh/io/io.h> 
+
  // GeoModel instantiation in 3D
  GeoModel3D geomodel ;
 
+ // give the path to your working directory
+ // give either the relative path from installation directory
+ // or an absolute path.
+ std::string path = "path/to/inputs_output_files/"
+
  // LOADING
- // path to the file to load
- // the file extention is the factory key.
- std::string input_file_name = "path/to/your_input_model.key";
+ // filename to load
+ // the file extension is the factory key.
+ std::string input_file_name = "your_input_model.key";
+ 
  // function to load a geomodel from input_file_name
- geomodel_load( geomodel, input_file_name ) ;
+ geomodel_load( geomodel, path + input_file_name ) ;
 
  // SAVING
  // path to the file to write
- // the file extention is the factory key.
- std::string output_file_name = "path/to/your_output_model.key";
- geomodel_save(geomodel, output_file_name);
+ // the file extension is the factory key.
+ std::string output_file_name = "your_output_model.key";
+ geomodel_save(geomodel, path + output_file_name);
 ```
 
 When a GeoModel is loaded, its validity is automatically checked and resulting information
 are displayed.
 
 ####Try:
 
- * Load the `ModelA1.ml` (a 3D model boundary representation from gocad).
- * Save it to the GINGMesh file format (.gm)`ModelA1.gm`.
+ * Load the `ModelA1.ml` (a 3D model boundary representation from gocad available in the `RINGMesh/data` directory).
+ * Save it to the RINGMesh file format (.gm)`ModelA1.gm`.
  * Open and look at the .gm file [see description](/features/file_formats).
 
 ### Basic Usage
 Several functions can be called to study/manipulate/repair the GeoModel:
 
 ```c++
 //print the geomodel statistics in the command terminal
+#include <ringmesh/geomodel/core/geomodel_api.h>
 print_geomodel_mesh_stats( geomodel ) ;
+
 //investigate the validity of the geomodel
+#include <ringmesh/geomodel/tools/geomodel_validity.h>
```

Comment:
Putting the includes here is better for understanding but this will not really compile... You cannot put includes everywhere. Not sure sure what is the best choice.